src/interactive/repl.ts

```diff
@@ -376,6 +376,63 @@ function clearProgress() {
     }
 }
 
+function display(params: { kind: string, data: any }) {
+    if (params.kind === 'application/vnd.julia-vscode.diagnostics') {
+        displayDiagnostics(params)
+    } else {
+        plots.displayPlot(params)
+    }
+}
+
+interface diagnosticData {
+    msg: string,
+    path: string,
+    line: number,
+    severity: number,
+    relatedInformation: {msg: string, path: string, line: number}[]
+}
+const g_trace_diagnostics: Map<string, vscode.DiagnosticCollection> = new Map()
+function displayDiagnostics(params: { kind: string, data: { source: string, items: diagnosticData[] } }) {
+    const source = params.data.source
+
+    if (g_trace_diagnostics.has(source)) {
+        g_trace_diagnostics.get(source).clear()
+    } else {
+        g_trace_diagnostics.set(source, vscode.languages.createDiagnosticCollection('Julia Diagnostics: ' + source))
```

Comment:
I would prefix this with something like `User provided Julia Diagnostics: ` or something like that, just to make clear that whatever message shows up here does not originate with us.

Reply:
As above, `Julia Runtime Diagnostics: ...` is probably most accurate, but this also doesn't show up anywhere.